core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -106,6 +105,7 @@ public void inject(AdvancedCache<Object, Object> cache, @ComponentName(CACHE_MAR
       this.persistenceExecutor = persistenceExecutor;
       this.byteBufferFactory = byteBufferFactory;
       this.marshalledEntryFactory = marshalledEntryFactory;
+      this.cacheStoreFactoryRegistry = cache.getComponentRegistry().getGlobalComponentRegistry().getComponent(CacheStoreFactoryRegistry.class);
```

Comment:
This isn't be a good approach, you shouldn't need to lookup `CacheStoreFactoryRegistry` component here, you simply have to add it as parameter in the `inject` method and the internals take of it.